pkg/prometheus/agent/operator_test.go

```diff
@@ -0,0 +1,201 @@
+package prometheusagent
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+)
+
+func TestValidateDaemonSetModeSpec(t *testing.T) {
```

Comment:
My preference would be to have the tables similar to what we have in https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/prometheus/server/operator_test.go#L41 to reduce the amount of boiler plate that we have. This is a little too verbose IMO.